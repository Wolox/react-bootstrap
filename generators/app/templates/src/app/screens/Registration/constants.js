export const FIELDS = {
  email: 'email',
  password: 'password'
};
