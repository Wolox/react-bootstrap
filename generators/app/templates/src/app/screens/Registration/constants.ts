import i18next from 'i18next';

import { Fields } from '~utils/types';

const personalData = {
  id: 'personalData',
  fields: [
    {
      name: 'firstName',
      label: i18next.t('Registration:firstName'),
      type: 'text',
      placeholder: i18next.t('Registration:firstNamePlaceholder'),
      required: true
    },
    {
      name: 'lastName',
      label: i18next.t('Registration:lastName'),
      type: 'text',
      placeholder: i18next.t('Registration:lastNamePlaceholder'),
      required: true
    }
  ]
};

const address = {
  id: 'address',
  fields: [
    {
      name: 'streetAndNumber',
      label: i18next.t('Registration:streetAndNumber'),
      type: 'text',
      placeholder: i18next.t('Registration:streetAndNumberPlaceholder')
    },
    {
      name: 'postCode',
      label: i18next.t('Registration:postCode'),
      type: 'text',
      placeholder: i18next.t('Registration:postCodePlaceholder')
    },
    {
      name: 'city',
      label: i18next.t('Registration:city'),
      type: 'text',
      placeholder: i18next.t('Registration:cityPlaceholder')
    }
  ]
};

const credentials = {
  id: 'credentials',
  fields: [
    {
      name: 'email',
      label: i18next.t('Registration:email'),
      type: 'email',
      placeholder: i18next.t('Registration:emailPlaceholder'),
      required: true
    },
    {
      name: 'password',
      label: i18next.t('Registration:password'),
      type: 'password',
      placeholder: i18next.t('Registration:passwordPlaceholder'),
      required: true
    },
    {
      name: 'confirmPassword',
      label: i18next.t('Registration:confirmPassword'),
      type: 'password',
      placeholder: i18next.t('Registration:confirmPasswordPlaceholder'),
      required: true
    }
  ]
};

export const steps: any[] = [personalData, address, credentials];

export const initialValues = steps.reduce(
  (acum, step) => ({
    ...acum,
    ...step.fields.reduce((acu: Fields, field: Fields) => ({ ...acu, [field.name]: field.initialValue }), {})
  }),
  {}
);
