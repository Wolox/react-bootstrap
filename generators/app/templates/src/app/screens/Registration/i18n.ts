import i18next from 'i18next';

i18next.addResources('es', 'Registration', {
  registration: 'Creación de cuenta',
  registrationExplanation: 'Ingrese los siguientes datos para crear su cuenta',
  name: 'Nombre',
  namePlaceholder: 'Ingrese su nombre',
  surname: 'Apellido',
  surnamePlaceholder: 'Ingrese su apellido',
  email: 'E-mail',
  emailPlaceholder: 'Ingrese su e-mail',
  password: 'Contraseña',
  passwordPlaceholder: 'Ingrese su contraseña',
  confirmPassword: 'Confirmar contraseña',
  confirmPasswordPlaceholder: 'Vuelva a ingresar su contraseña',
  required: '* Requerido',
  submit: 'Enviar'
});
