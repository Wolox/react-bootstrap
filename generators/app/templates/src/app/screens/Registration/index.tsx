import React, { useCallback } from 'react';
import i18next from 'i18next';

import FormInput from '~components/FormInput';
import PATHS from '~components/Routes/paths';
import { stringArrayToObject } from '~utils/array';

import styles from './styles.module.scss';

const FIELDS = stringArrayToObject(['name', 'surname', 'email', 'password', 'confirmPassword']);

function RegistrationContainer() {
  /* eslint-disable @typescript-eslint/no-unused-vars */
  const handleSubmit = useCallback((event: React.FormEvent<Element>) => {
    // TODO: Implement
  }, []);

  const handleInputChange = useCallback((event: React.FormEvent<Element>) => {
    // TODO: Implement method
  }, []);

  return (
    <form className={`column center full-width ${styles.formContainer}`} onSubmit={handleSubmit}>
      <div className="column center m-bottom-3">
        <h1 className="m-bottom-1">{i18next.t('Registration:registration')}</h1>
        <h2>{i18next.t('Registration:registrationExplanation')}</h2>
      </div>
      <div className={`row space-between ${styles.sectionContainer}`}>
        <FormInput
          className={styles.inputContainer}
          label={i18next.t('Registration:name')}
          name={FIELDS.name}
          inputType="text"
          inputClassName={styles.input}
          placeholder={i18next.t('Registration:namePlaceholder') as string}
          onChange={handleInputChange}
        />
        <FormInput
          className="full-width m-left-1"
          label={i18next.t('Registration:surname')}
          name={FIELDS.surname}
          inputType="text"
          inputClassName={styles.input}
          placeholder={i18next.t('Registration:surnamePlaceholder') as string}
          onChange={handleInputChange}
        />
      </div>
      <div className={`row ${styles.sectionContainer}`}>
        <FormInput
          className={styles.inputContainer}
          label={i18next.t('Registration:email')}
          name={FIELDS.email}
          inputType="text"
          inputClassName={styles.input}
          placeholder={i18next.t('Registration:emailPlaceholder') as string}
          onChange={handleInputChange}
        />
      </div>
      <div className={`row space-between ${styles.sectionContainer}`}>
        <FormInput
          className={styles.inputContainer}
          label={i18next.t('Registration:password')}
          name={FIELDS.password}
          inputType="password"
          inputClassName={styles.input}
          placeholder={i18next.t('Registration:passwordPlaceholder') as string}
          onChange={handleInputChange}
        />
        <FormInput
          className="full-width m-left-1"
          label={i18next.t('Registration:confirmPassword')}
          name={FIELDS.confirmPassword}
          inputType="password"
          inputClassName={styles.input}
          placeholder={i18next.t('Registration:confirmPasswordPlaceholder') as string}
          onChange={handleInputChange}
        />
      </div>
      <div className={`column center ${styles.sectionContainer}`}>
        <button type="submit" className={`full-width m-bottom-1 ${styles.button}`}>
          {i18next.t('RecoverPassword:enter')}
        </button>
        <a href={PATHS.login}>{i18next.t('RecoverPassword:returnToLogin')}</a>
      </div>
    </form>
  );
}

export default RegistrationContainer;
