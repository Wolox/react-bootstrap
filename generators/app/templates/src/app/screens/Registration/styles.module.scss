.button {
  border: solid 1px;
  border-radius: 5px;
  height: 40px;
}

.input-container {
  padding: 5px 0;
}

.input {
  border: solid 1px;
  border-radius: 5px;
  font-size: 14px;
  height: 30px;
  padding: 0 5px;
  width: 100%;
}

.form-container {
  min-height: 500px;
  padding-top: 80px;
}

.section-container {
  max-width: 450px;
  margin-bottom: 20px;
  padding-right: 10px;
  width: 100%;
}
