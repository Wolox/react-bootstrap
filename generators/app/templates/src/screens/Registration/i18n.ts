import i18next from 'i18next';

i18next.addResources('es', 'Registration', {
  registration: 'Creación de cuenta',
  registrationExplanation: 'Ingrese los siguientes datos para crear su cuenta',
  firstName: 'Nombre',
  firstNamePlaceholder: 'Ingrese su nombre',
  lastName: 'Apellido',
  lastNamePlaceholder: 'Ingrese su apellido',
  email: 'E-mail',
  emailPlaceholder: 'Ingrese su e-mail',
  password: 'Contraseña',
  passwordPlaceholder: 'Ingrese su contraseña',
  confirmPassword: 'Confirmar contraseña',
  confirmPasswordPlaceholder: 'Vuelva a ingresar su contraseña',
  required: '* Requerido',
  submit: 'Enviar',
  emailFormatError: 'El formato es inválido',
  requiredError: 'El campo es requerido',
  confirmPasswordError: 'Las contraseñas no coinciden',
  passwordLengthError: 'Debe tener al menos {{amount}} caracteres'
});
