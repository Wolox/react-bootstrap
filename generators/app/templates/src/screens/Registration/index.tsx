import React, { useCallback } from 'react';
import i18next from 'i18next';
import { useHistory } from 'react-router';
import { useForm } from 'react-hook-form';

import FormInput from 'components/FormInput';
import PATHS from 'components/Routes/paths';
import { useDispatch } from 'contexts/UserContext';
import { useLazyRequest } from 'hooks/useRequest';
import { signup, setCurrentUser, RegistrationUser } from 'services/AuthServices';
import { actionCreators, User } from 'contexts/UserContext/reducer';

import styles from './styles.module.scss';

const PASSWORD_LENGTH = 8;

const FIELDS = {firstName: 'firstName', lastName: 'lastName', email: 'email', password: 'password', confirmPassword: 'confirmPassword'};

const requiredValidation = { required: i18next.t('Registration:requiredError') as string };

const VALIDATIONS = {
  firstName: requiredValidation,
  lastName: requiredValidation,
  email: {
    pattern: {
      value: /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
      message: i18next.t('Registration:emailFormatError') as string
    },
    ...requiredValidation
  },
  password: {
    ...requiredValidation,
    minLength: {
      value: 8,
      message: i18next.t('Registration:passwordLengthError', { amount: PASSWORD_LENGTH }) as string
    }
  },
  confirmPassword: requiredValidation
};

function RegistrationContainer() {
  const { register, handleSubmit, errors, watch } = useForm();
  const dispatch = useDispatch();
  const history = useHistory();

  const [, , , signupRequest] = useLazyRequest({
    request: signup,
    withPostSuccess: response => {
      const userResponse = response as User;
      dispatch(actionCreators.setUser(userResponse));
      setCurrentUser(userResponse);

      history.push('/');
    }
  });

  const handleRegister = useCallback(
    (values: RegistrationUser) => {
      signupRequest(values);
    },
    [signupRequest]
  );

  const validateConfirmPassword = (value: string) =>
    value === watch('password') || (i18next.t('Registration:confirmPasswordError') as string);

  return (
    <form
      className={`column center full-width ${styles.formContainer}`}
      aria-label="registration-form"
      onSubmit={handleSubmit(handleRegister)}
      noValidate
    >
      <div className="column center m-bottom-3">
        <h1 className="m-bottom-1">{i18next.t('Registration:registration')}</h1>
        <h2>{i18next.t('Registration:registrationExplanation')}</h2>
      </div>
      <div className={`row space-between ${styles.sectionContainer}`}>
        <FormInput
          className={`${styles.inputContainer} item-1 m-right-2`}
          label={i18next.t('Registration:firstName')}
          name={FIELDS.firstName}
          inputType="text"
          inputClassName={styles.input}
          placeholder={i18next.t('Registration:firstNamePlaceholder') as string}
          inputRef={register(VALIDATIONS.firstName)}
          error={errors?.firstName?.message}
        />
        <FormInput
          className={`${styles.inputContainer} item-1`}
          label={i18next.t('Registration:lastName')}
          name={FIELDS.lastName}
          inputType="text"
          inputClassName={styles.input}
          placeholder={i18next.t('Registration:lastNamePlaceholder') as string}
          inputRef={register(VALIDATIONS.lastName)}
          error={errors?.lastName?.message}
        />
      </div>
      <div className={`row ${styles.sectionContainer}`}>
        <FormInput
          className={`${styles.inputContainer} item-1`}
          label={i18next.t('Registration:email')}
          name={FIELDS.email}
          inputType="email"
          inputClassName={styles.input}
          placeholder={i18next.t('Registration:emailPlaceholder') as string}
          inputRef={register(VALIDATIONS.email)}
          error={errors?.email?.message}
        />
      </div>
      <div className={`row space-between ${styles.sectionContainer}`}>
        <FormInput
          className={`${styles.inputContainer} item-1 m-right-2`}
          label={i18next.t('Registration:password')}
          name={FIELDS.password}
          inputType="password"
          inputClassName={styles.input}
          placeholder={i18next.t('Registration:passwordPlaceholder') as string}
          inputRef={register(VALIDATIONS.password)}
          error={errors?.password?.message}
        />
        <FormInput
          className={`${styles.inputContainer} item-1`}
          label={i18next.t('Registration:confirmPassword')}
          name={FIELDS.confirmPassword}
          inputType="password"
          inputClassName={styles.input}
          placeholder={i18next.t('Registration:confirmPasswordPlaceholder') as string}
          inputRef={register({ ...VALIDATIONS.confirmPassword, validate: validateConfirmPassword })}
          error={errors?.confirmPassword?.message}
        />
      </div>
      <div className={`column center ${styles.sectionContainer}`}>
        <button type="submit" className={`full-width m-bottom-1 ${styles.button}`}>
          {i18next.t('RecoverPassword:enter')}
        </button>
        <a href={PATHS.login}>{i18next.t('RecoverPassword:returnToLogin')}</a>
      </div>
    </form>
  );
}

export default RegistrationContainer;
