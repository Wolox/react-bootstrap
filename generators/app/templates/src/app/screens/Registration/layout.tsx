import React from 'react';
import i18next from 'i18next';

import InputLabel from '/home/trocchi/Documents/frontend/test/login/src/app/components/InputLabel';

import Routes from '/home/trocchi/Documents/frontend/test/login/src/constants/routes';

import { FIELDS } from './constants';
import styles from './styles.module.scss';

interface Props {
  onEmailChange: (event: React.FormEvent<Element>) => void;
  onPasswordChange: (event: React.FormEvent<Element>) => void;
  onSubmit: (event: React.FormEvent<HTMLFormElement>) => void;
}

function Registration({ onEmailChange, onPasswordChange, onSubmit }: Props) {
  return (
    <form className={`column center full-width m-top-8 ${styles.formContainer}`} onSubmit={onSubmit}>
      <div className="column center m-bottom-3">
        <h1 className="m-bottom-1">{i18next.t('Registration:registration')}</h1>
        <h2>{i18next.t('Registration:registrationExplanation')}</h2>
      </div>
      <div className={`column m-bottom-2 ${styles.sectionContainer}`}>
        {//FIELDS.map(field => )}
}
      </div>
      <div className={`column center ${styles.sectionContainer}`}>
        <button type="submit" className={`full-width m-bottom-1 ${styles.button}`}>
          {i18next.t('Registration:enter')}
        </button>
        <a href={Routes.RECOVER_PASSWORD}>{i18next.t('Registration:forgotPassword')}</a>
      </div>
    </form>
  );
}

export default Registration;
