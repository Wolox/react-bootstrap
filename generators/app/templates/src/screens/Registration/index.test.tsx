import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';

import Registration from './index';

const mockSetStateUser = jest.fn();
const mockSetPersistantUser = jest.fn();

jest.mock('~contexts/UserContext/reducer', () => ({
  actionCreators: {
    setUser: (values: any) => mockSetStateUser(values)
  }
}));

jest.mock('~services/AuthServices', () => ({
  signup: () =>
    new Promise(resolve =>
      resolve({
        ok: true,
        data: { sessionToken: 'token', id: 1234 },
        problem: null,
        originalError: null
      })
    ),
  setCurrentUser: (values: any) => mockSetPersistantUser(values)
}));

describe('#Registration', () => {
  const component = (
    <Router history={createMemoryHistory()}>
      <Registration />
    </Router>
  );

  const validValues = {
    firstName: 'John',
    lastName: 'Doe',
    email: 'someEmail@wolox.com',
    password: 'aPassword'
  };

  describe('when mounting', () => {
    it('shows valid content', () => {
      const { container } = render(component);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('when values are empty and form is submitted', () => {
    it('shows the required message for each required field', async () => {
      const { getByRole, getByLabelText } = render(component);
      const firstName = await getByLabelText('firstName');
      const lastName = await getByLabelText('lastName');
      const email = await getByLabelText('email');
      const password = await getByLabelText('password');
      const passwordConfirmation = await getByLabelText('confirmPassword');
      const form = await getByRole('form', { name: 'registration-form' });

      // eslint-disable-next-line max-nested-callbacks
      await waitFor(() => fireEvent.submit(form));

      expect(firstName.parentElement?.innerHTML).toMatch('requiredError');
      expect(lastName.parentElement?.innerHTML).toMatch('requiredError');
      expect(email.parentElement?.innerHTML).toMatch('requiredError');
      expect(password.parentElement?.innerHTML).toMatch('requiredError');
      expect(passwordConfirmation.parentElement?.innerHTML).toMatch('requiredError');
    });
  });

  describe('when email format is invalid and form is submitted', () => {
    it('shows the email error', async () => {
      const { getByRole, getByLabelText, getAllByRole } = render(component);
      const firstName = await getByLabelText('firstName');
      const lastName = await getByLabelText('lastName');
      const email = await getByLabelText('email');
      const password = await getByLabelText('password');
      const passwordConfirmation = await getByLabelText('confirmPassword');
      const form = await getByRole('form', { name: 'registration-form' });

      await fireEvent.change(firstName, { target: { value: validValues.firstName } });
      await fireEvent.change(lastName, { target: { value: validValues.lastName } });
      await fireEvent.change(email, { target: { value: 'invalid email' } });
      await fireEvent.change(password, { target: { value: validValues.password } });
      await fireEvent.change(passwordConfirmation, { target: { value: validValues.password } });

      // eslint-disable-next-line max-nested-callbacks
      await waitFor(() => fireEvent.submit(form));
      const errors = await getAllByRole('alert');

      expect(email.parentElement?.innerHTML).toMatch('emailFormatError');
      expect(errors.length).toBe(1);
    });
  });

  describe('when password length is too short and form is submitted', () => {
    it('shows the password error', async () => {
      const { getByRole, getByLabelText, getAllByRole } = render(component);
      const firstName = await getByLabelText('firstName');
      const lastName = await getByLabelText('lastName');
      const email = await getByLabelText('email');
      const password = await getByLabelText('password');
      const passwordConfirmation = await getByLabelText('confirmPassword');
      const form = await getByRole('form', { name: 'registration-form' });

      await fireEvent.change(firstName, { target: { value: validValues.firstName } });
      await fireEvent.change(lastName, { target: { value: validValues.lastName } });
      await fireEvent.change(email, { target: { value: validValues.email } });
      await fireEvent.change(password, { target: { value: 'short' } });
      await fireEvent.change(passwordConfirmation, { target: { value: 'short' } });

      // eslint-disable-next-line max-nested-callbacks
      await waitFor(() => fireEvent.submit(form));
      const errors = await getAllByRole('alert');

      expect(password.parentElement?.innerHTML).toMatch('passwordLengthError');
      expect(errors.length).toBe(1);
    });
  });

  describe('when password confirmation does not match password and form is submitted', () => {
    it('shows the password confirmation error', async () => {
      const { getByRole, getByLabelText, getAllByRole } = render(component);
      const firstName = await getByLabelText('firstName');
      const lastName = await getByLabelText('lastName');
      const email = await getByLabelText('email');
      const password = await getByLabelText('password');
      const passwordConfirmation = await getByLabelText('confirmPassword');
      const form = await getByRole('form', { name: 'registration-form' });

      await fireEvent.change(firstName, { target: { value: validValues.firstName } });
      await fireEvent.change(lastName, { target: { value: validValues.lastName } });
      await fireEvent.change(email, { target: { value: validValues.email } });
      await fireEvent.change(password, { target: { value: validValues.password } });
      await fireEvent.change(passwordConfirmation, { target: { value: `${validValues.password}a` } });

      // eslint-disable-next-line max-nested-callbacks
      await waitFor(() => fireEvent.submit(form));

      const errors = await getAllByRole('alert');

      expect(passwordConfirmation.parentElement?.innerHTML).toMatch('confirmPasswordError');
      expect(errors.length).toBe(1);
    });
  });

  describe('when values are valid and form is submitted', () => {
    it('created the user', async () => {
      const { getByRole, getByLabelText } = render(component);
      const firstName = await getByLabelText('firstName');
      const lastName = await getByLabelText('lastName');
      const email = await getByLabelText('email');
      const password = await getByLabelText('password');
      const passwordConfirmation = await getByLabelText('confirmPassword');
      const form = await getByRole('form', { name: 'registration-form' });

      await fireEvent.change(firstName, { target: { value: validValues.firstName } });
      await fireEvent.change(lastName, { target: { value: validValues.lastName } });
      await fireEvent.change(email, { target: { value: validValues.email } });
      await fireEvent.change(password, { target: { value: validValues.password } });
      await fireEvent.change(passwordConfirmation, { target: { value: validValues.password } });

      // eslint-disable-next-line max-nested-callbacks
      await waitFor(() => fireEvent.submit(form));

      expect(mockSetStateUser).toHaveBeenCalled();
      expect(mockSetStateUser).toHaveBeenCalledWith({ sessionToken: 'token', id: 1234 });
      expect(mockSetPersistantUser).toHaveBeenCalled();
      expect(mockSetPersistantUser).toHaveBeenCalledWith({ sessionToken: 'token', id: 1234 });
    });
  });
});
