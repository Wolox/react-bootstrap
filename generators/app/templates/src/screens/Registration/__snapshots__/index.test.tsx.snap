// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Registration when mounting shows valid content 1`] = `
<form
  aria-label="registration-form"
  class="column center full-width formContainer"
  novalidate=""
>
  <div
    class="column center m-bottom-3"
  >
    <h1
      class="m-bottom-1"
    >
      registration
    </h1>
    <h2>
      registrationExplanation
    </h2>
  </div>
  <div
    class="row space-between sectionContainer"
  >
    <div
      class="column start inputContainer item-1 m-right-2"
    >
      <label
        class=" m-bottom-1"
        for="firstName"
      >
        firstName
      </label>
      <input
        class="input input "
        id="firstName"
        name="firstName"
        placeholder="firstNamePlaceholder"
        type="text"
      />
      <span
        aria-hidden="true"
        class=" errorText "
        id="firstName-error"
        role="alert"
      />
    </div>
    <div
      class="column start inputContainer item-1"
    >
      <label
        class=" m-bottom-1"
        for="lastName"
      >
        lastName
      </label>
      <input
        class="input input "
        id="lastName"
        name="lastName"
        placeholder="lastNamePlaceholder"
        type="text"
      />
      <span
        aria-hidden="true"
        class=" errorText "
        id="lastName-error"
        role="alert"
      />
    </div>
  </div>
  <div
    class="row sectionContainer"
  >
    <div
      class="column start inputContainer item-1"
    >
      <label
        class=" m-bottom-1"
        for="email"
      >
        email
      </label>
      <input
        class="input input "
        id="email"
        name="email"
        placeholder="emailPlaceholder"
        type="email"
      />
      <span
        aria-hidden="true"
        class=" errorText "
        id="email-error"
        role="alert"
      />
    </div>
  </div>
  <div
    class="row space-between sectionContainer"
  >
    <div
      class="column start inputContainer item-1 m-right-2"
    >
      <label
        class=" m-bottom-1"
        for="password"
      >
        password
      </label>
      <input
        class="input input "
        id="password"
        name="password"
        placeholder="passwordPlaceholder"
        type="password"
      />
      <span
        aria-hidden="true"
        class=" errorText "
        id="password-error"
        role="alert"
      />
    </div>
    <div
      class="column start inputContainer item-1"
    >
      <label
        class=" m-bottom-1"
        for="confirmPassword"
      >
        confirmPassword
      </label>
      <input
        class="input input "
        id="confirmPassword"
        name="confirmPassword"
        placeholder="confirmPasswordPlaceholder"
        type="password"
      />
      <span
        aria-hidden="true"
        class=" errorText "
        id="confirmPassword-error"
        role="alert"
      />
    </div>
  </div>
  <div
    class="column center sectionContainer"
  >
    <button
      class="full-width m-bottom-1 button"
      type="submit"
    >
      submit
    </button>
    <a
      href="/login"
    >
      returnToLogin
    </a>
  </div>
</form>
`;
